File: prd-ingestion-front/src/models/template.ts
// ============================================================
// 模板管理相关类型
// ============================================================

import type { DocumentType } from './prd'

/** 模板版本 */
export type TemplateVersion = {
  version: number
  content: string
  updatedAt: string
  updatedBy: string
  changeLog: string
}

/** 模板详情 */
export type Template = {
  templateId: string
  type: DocumentType
  name: string
  description: string
  content: string // Markdown 模板，含 {{PLACEHOLDER}} 变量
  version: number
  versions: TemplateVersion[]
  updatedAt: string
  updatedBy: string
}

/** 模板创建请求 */
export type TemplateCreateRequest = {
  type: DocumentType
  name: string
  description: string
  content: string
}

/** 模板更新请求 */
export type TemplateUpdateRequest = {
  content: string
  changeLog: string
}

File: prd-ingestion-server/src/main/java/com/tayama/prd/ingestion/model/template/Template.java
package com.tayama.prd.ingestion.model.template;

import java.util.List;

/**
 * 模板详情。
 */
public class Template {
    private String templateId;
    private String type;
    private String name;
    private String description;
    private String content;
    private int version;
    private List<TemplateVersion> versions;
    private String updatedAt;
    private String updatedBy;

    public String getTemplateId() { return templateId; }
    public void setTemplateId(String templateId) { this.templateId = templateId; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public int getVersion() { return version; }
    public void setVersion(int version) { this.version = version; }
    public List<TemplateVersion> getVersions() { return versions; }
    public void setVersions(List<TemplateVersion> versions) { this.versions = versions; }
    public String getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }
    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }

    public static class TemplateVersion {
        private int version;
        private String content;
        private String updatedAt;
        private String updatedBy;
        private String changeLog;

        public int getVersion() { return version; }
        public void setVersion(int version) { this.version = version; }
        public String getContent() { return content; }
        public void setContent(String content) { this.content = content; }
        public String getUpdatedAt() { return updatedAt; }
        public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }
        public String getUpdatedBy() { return updatedBy; }
        public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
        public String getChangeLog() { return changeLog; }
        public void setChangeLog(String changeLog) { this.changeLog = changeLog; }
    }
}